### PR TITLE
Revive Svg.Set

### PR DESCRIPTION
**Pull request by Fuzzyma**
This PR brings back SVG.Set which was removed earlier.
It is now implemented as what is called an array-like object. This is nothing more than an object with numeral properties and a length property.
This has the benefit that we can use all array methods on it and it will just work.
It is designed after the proposal #645. The map/collect method is missing atm but can be easily added later.

### CODE REVIEW

**Comment by saivan** [comment]
Is there a reason not to just use the built in es6 `Set` and include a polyfill?

File: src/set.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      // return an empty box of there are no members
+      if (this.length == 0)
+        return new SVG.Box()
+
+      // get the first rbox and update the target bbox
+      var rbox = this[0].rbox(this[0].doc())
+
+      this.each(function() {
+        // user rbox for correct position and visual representation
+        rbox = rbox.merge(this.rbox(this.doc()))
```
Comment (saivan) [inline src/set.js]: Why not store a reference to this.doc() above each?

File: src/set.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  , remove: function(element) {
+      var i = this.index(element)
+
+      // remove given child
+      if (i > -1)
+        Array.prototype.splice.call(this, i, 1)
+
+      return this
+    }
+  , each: function (block, value) {
```
Comment (saivan) [inline src/set.js]: Its unclear to me what block should be if it isn't a function?
Can you clarify?

File: src/set.js (hunk 1 above)
Comment (Fuzzyma) [reply src/set.js]: Good question. Will change that!

File: src/set.js (hunk 2 above)
Comment (Fuzzyma) [reply src/set.js]: it can be a property of an element, too. I will rename it to property to make the intention more clear

**Comment by saivan** [comment]
Does this need changes now because of the discussion we had over the phone?

**Comment by Fuzzyma** [comment]
Jeah the question is if we keep the inherit stuff. It sounded like you had a strong opinion about its usefulness. And I agree that the each solution can be quite ugly.
So we might change it

**Comment by saivan** [comment]
Yep, I mean; I think it is very useful, as we discussed; but perhaps we should discuss some more once we have better solutions to keep it 😃

**Pull request closed without merging**